### PR TITLE
Add docs for map_to_list processor

### LINKED ISSUE #6132: [DOC] Add doc for Map to List processor in Data Prepper
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Add new docs for a new processor - `map_to_list` processor for Data Prepper.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/data-prepper/pull/3945

### PR DESCRIPTION
**Pull request by oeyh**
Closes [#6132 ](https://github.com/opensearch-project/documentation-website/issues/6132)

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
@dlvenable: Can you take a look at this one?

**Review by Naarcha-AWS — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

File: _data-prepper/pipelines/configuration/processors/map-to-list.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    "key1": "value1",
+    "key2": "value2",
+    "key3": "value3"
+  }
+}
+```
+
+### Example: Use the event root as the source
+
+The following example shows how you can use an event's root as the source by setting the `source` setting to empty string (`""`).:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/map-to-list.md:233]:
```suggestion
The following example shows how you can use an event's root as the source by setting the `source` setting to an empty string (`""`):
```

File: _data-prepper/pipelines/configuration/processors/mutate-event.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ Mutate event processors allow you to modify events in Data Prepper. The followin
 * [rename_keys]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/rename-keys/) allows you to rename keys in an event.
 * [convert_entry_type]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/convert_entry_type/) allows you to convert value types in an event.
 * [list_to_map]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/list-to-map) allows you to convert list of objects from an event where each object contains a `key` field into a map of target keys.
+* [map_to_list]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/map-to-list) allows you to converts a map of key-value pairs to a list of objects, each contains the key and value in separate fields.q
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/mutate-event.md:19]:
```suggestion
* [map_to_list]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/map-to-list) allows you to convert a map of key-value pairs to a list of objects, with each object containing the key and value in separate fields.
```

**Review by natebower — COMMENTED** [review]
@oeyh @Naarcha-AWS Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**